src/main/java/io/cdap/plugin/gcp/gcs/actions/GCSBucketCreate.java

```diff
@@ -192,11 +193,29 @@ public boolean failIfExists() {
 
     void validate(FailureCollector collector) {
       if (!containsMacro("paths")) {
+        Storage storage = null;
+        try {
+          Credentials credentials = getServiceAccountFilePath() == null ?
+            null : GCPUtils.loadServiceAccountCredentials(getServiceAccountFilePath());
+          storage = GCPUtils.getStorage(getProject(), credentials);
+        } catch (IOException e) {
+          collector.addFailure(e.getMessage(), "Ensure you entered the correct file path.")
+            .withConfigProperty(NAME_SERVICE_ACCOUNT_FILE_PATH)
+            .withStacktrace(e.getStackTrace());
+        }
         for (String path : getPaths()) {
           try {
-            GCSPath.from(path);
+            GCSPath gcsPath = GCSPath.from(path);
+            // Check if bucket exists and is accessible
+            if (storage != null) {
+              storage.get(gcsPath.getBucket());
+            }
           } catch (IllegalArgumentException e) {
             collector.addFailure(e.getMessage(), null).withConfigElement(NAME_PATHS, path);
+          } catch (StorageException e) {
+            collector.addFailure(e.getMessage(), "Ensure you entered the correct bucket path.")
```

Comment:
Is `StorageException` only thrown when bucket path is not correct?

Reply:
I believe that StorageException is simply an exception which encapsulates all possible API responses from the GCS API. However, this is outdated since we are moving these checks to happen only at runtime. In most cases, an exception happens due to lack of permissions to access a bucket (403). See https://cloud.google.com/storage/docs/json_api/v1/status-codes

Reply:
Although the permissions error is not the only message which can occur as a result of the StorageException, it is the most likely response assuming GCS doesn't throw a 500 or something. Another option is to parse the response error code and use that instead to determine when to throw a more detailed error message.